pytorch_to_returnn/torch/nn/modules/linear.py

```diff
@@ -60,6 +60,27 @@ def import_params_torch_to_returnn(self, *, layer: LinearLayer, torch_module: Li
       layer.params["b"].load(torch_module.bias.detach().numpy(), session=session)
 
 
+class FunctionalLinear(Module):
+  is_original_torch_module = False
+
+  def __init__(self):
+    super(FunctionalLinear, self).__init__()
+
+  def create_returnn_layer_dict(self, input: Tensor, weight: Tensor, bias: Optional[Tensor] = None) -> Dict[str, Any]:
+    assert len(weight.shape) == 2
+    out_features, in_features = weight.shape
+    if bias is not None:
+      assert bias.shape == (out_features,)
+    return {
+      "class": "linear", "from": self._get_input_layer_name(input),
+      "n_out": out_features,
+      "with_bias": bias is not None,
+      "bias": self._get_input_layer_name(bias) if bias is not None else None,
+      "weights": self._get_input_layer_name(weight),
+      "use_transposed_weights": True,
+      "activation": None}
```

Comment:
Rather than defining `create_returnn_layer_dict`, you should define `forward` and use `functional.matmul` and just `+` for the bias.

Reply:
True, but then actually we don't need this module at all, right? We can just do it in `functional.linear`.

Reply:
Yes. This is maybe a matter of taste. In one case, it will end up in a subnetwork, in the other not.

Reply:
This still should be cleaned up. I think you don't need it at all anymore, right?

Reply:
That's right.